org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WRecordIDEditor.java

```diff
@@ -283,9 +299,14 @@ private void setValue (Object value, boolean fire) {
 				tableIDValue = tableIDGridField.getValue();
 			}
 			if(value != null && tableIDValue != null) {
-				int recordID = Integer.parseInt(String.valueOf(value));
 				int tableID = Integer.parseInt(String.valueOf(tableIDValue));
-				if(recordID >= 0 && tableID > 0)
+				MTable table = MTable.get(tableID);
```

Comment:
again, if tableID can be zero, I think it is better to wrap this inside a if (tableID > 0) condition block.

Reply:
done and tested